sites/shared/components/inputs.mjs

```diff
@@ -0,0 +1,424 @@
+// Dependencies
+import { cloudflareImageUrl } from 'shared/utils.mjs'
+import { collection } from 'shared/hooks/use-design.mjs'
+// Context
+import { ModalContext } from 'shared/context/modal-context.mjs'
+// Hooks
+import { useState, useCallback, useContext } from 'react'
+import { useTranslation } from 'next-i18next'
+import { useDropzone } from 'react-dropzone'
+// Components
+import { Popout } from 'shared/components/popout/index.mjs'
+import Markdown from 'react-markdown'
+import { ResetIcon, DocsIcon, HelpIcon } from 'shared/components/icons.mjs'
+import { ModalWrapper } from 'shared/components/wrappers/modal.mjs'
+import { isDegreeMeasurement } from 'config/measurements.mjs'
+import { measurementAsMm, formatMm, measurementAsUnits, parseDistanceInput } from 'shared/utils.mjs'
+
+//import { Collapse } from 'shared/components/collapse.mjs'
+//import { PlusIcon, EditIcon } from 'shared/components/icons.mjs'
+//import { NumberInput } from 'shared/components/workbench/menus/shared/inputs.mjs'
+//import { useState, useCallback } from 'react'
+
+export const ns = ['account', 'measurements', 'designs']
+
+/*
+ * Helper component to display a tab heading
+ */
+export const Tab = ({
+  id, // The tab ID
+  label, // A label for the tab, if not set we'll use the ID
+  activeTab, // Which tab (id) is active
+  setActiveTab, // Method to set the active tab
+}) => (
+  <button
+    className={`text-lg font-bold capitalize tab tab-bordered grow
+    ${activeTab === id ? 'tab-active' : ''}`}
+    onClick={() => setActiveTab(id)}
+  >
+    {label ? label : id}
+  </button>
+)
+
+/*
+ * Helper component to wrap a form control with a label
+ */
+export const FormControl = ({
+  label, // the (top-left) label
+  children, // Children to go inside the form control
+  docs = false, // Optional top-right label
+  labelBL = false, // Optional bottom-left label
+  labelBR = false, // Optional bottom-right label
+}) => {
+  const { setModal } = useContext(ModalContext)
+
+  return (
+    <div className="form-control w-full mt-2">
+      <label className={`label pb-0 ${docs ? 'hover:cursor-help' : ''}`}>
+        <span className="label-text text-lg font-bold mb-0">{label}</span>
+        {docs ? (
+          <span className="label-text-alt">
+            <button
+              className="btn btn-ghost btn-sm btn-circle hover:btn-secondary"
+              onClick={() =>
+                setModal(
+                  <ModalWrapper flex="col" justify="top lg:justify-center" slideFrom="right">
+                    {docs}
+                  </ModalWrapper>
+                )
+              }
+            >
+              <DocsIcon />
+            </button>
+          </span>
+        ) : null}
+      </label>
+      {children}
+      {labelBL || labelBR ? (
+        <label className="label">
```

Comment:
I have re-worked the FormControl component so that you can pass an `id` prop to the various inputs.
If you do so, the FormComponent will link the label with the input with id/for.
If you do not, it will render a div instead of a label.

While I did all the work to make it possible, I have not (yet) passed an ID to all inputs. But it's at least possible now.